conanfile.py

```diff
@@ -2,25 +2,14 @@
 
 
 class CppStarterProject(ConanFile):
-    # Note: options are copied from CMake boolean options.
-    # When turned off, CMake sometimes passes them as empty strings.
-    options = {
-        "cpp_starter_use_imgui": ["ON", "OFF", ""],
-        "cpp_starter_use_sdl": ["ON", "OFF", ""]
-    }
     name = "CppStarterProject"
     version = "0.1"
     requires = (
         "catch2/2.13.7",
         "docopt.cpp/0.6.2",
         "fmt/8.0.1",
         "spdlog/1.9.2",
+        # "imgui-sfml/2.1@bincrafters/stable"
+        # "sdl2/2.0.10@bincrafters/stable"
```

Comment:
I know you're trying to make `cmakelib` more reusable, and that means not having variables like `CPP_STARTER_USE_IMGUI` and `CPP_STARTER_USE_SDL` in it, but there must be some other way to set options that will be passed directly to `conan_cmake_install`. Perhaps the consumer of `cmakelib` can set a `CONAN_OPTIONS` variable and call `cmakelib(... EXTRA_CONAN_OPTIONS ${CONAN_OPTIONS}`, and then `cmakelib()` can call `conan_cmake_install(... ${EXTRA_CONAN_OPTIONS})`?

Have you considered making this change as a separate PR?

If we are going this direction, I think it would make more sense to replace the whole `conanfile.py` with a `conanfile.txt`, and then users can uncomment `imgui` and `sdl2` as needed. As is, this change makes it impossible to install `imgui` and `sdl2` when you run the CMake configure step, and that could be important for testing the sdl and imgui subprojects in CI. This goes against the ["one step sets up everything" ideal](https://github.com/cpp-best-practices/cpp_starter_project/issues/147#issuecomment-904277057) that @lefticus was going for initially.

Personally, I don't have a problem with this. At best, the conan `sdl2` package takes forever to build and it usually doesn't work anyway. The situation with `imgui-sfml` is much better, but I think the overhead involved in building it is very high compared to just installing a prebuilt binary. In CI at least, I think it's a big improvement to use a package manager for this rather than build these dependencies with Conan.

Reply:
Fixed in https://github.com/cpp-best-practices/cpp_starter_project/pull/169/commits/e0d227fd6952bf0f96083ad4365e751a2af4af80

Reply:
I prefer that we remove all the unnecessary and fragile dynamism from the codebase. At the end of the day, people want to decide on one or more libraries and only use those. There is no need to add global mutable options that make the build fragile and unnecessarily dynamic.

Reply:
In that case, please replace this file with a `conanfile.txt`. After these changes, there is no benefit to using a `conanfile.py`.

Reply:
Fixed in https://github.com/cpp-best-practices/cpp_starter_project/pull/169/commits/e6b4af38ab46df8f840824b1f05b9fc3d6e7597b